python-sdk/nuscenes/tests/test_predict_helper.py

```diff
@@ -0,0 +1,503 @@
+# nuScenes dev-kit.
+# Code written by Freddy Boulton, 2020.
+
+import unittest
+from typing import Dict, List, Any
+from nuscenes.predict import PredictHelper, convert_global_coords_to_local
+import numpy as np
+
+class MockNuScenes:
+    """Mocks the NuScenes API needed to test PredictHelper"""
+
+    def __init__(self, sample_annotations: List[Dict[str, Any]],
+                 samples: List[Dict[str, Any]]):
+
+        self._sample_annotation = {r['token']: r for r in sample_annotations}
+        self._sample = {r['token']: r for r in samples}
+
+    @property
+    def sample_annotation(self,) -> List[Dict[str, Any]]:
+        return list(self._sample_annotation.values())
+
+    def get(self, table_name: str, token: str) -> Dict[str, Any]:
+        assert table_name in {'sample_annotation', 'sample'}
+        return getattr(self, "_" + table_name)[token]
+
+class Test_convert_global_coords_to_local(unittest.TestCase):
+
+    def setUp(self):
+        along_pos_x = np.zeros((5, 2))
```

Comment:
Some comments on what these are may be helpful.

Reply:
Done!